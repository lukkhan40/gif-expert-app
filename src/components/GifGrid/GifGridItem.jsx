import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => (
  <div className="card">
    <img alt={title} src={url} />
    <p>{title}</p>
  </div>
);

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
