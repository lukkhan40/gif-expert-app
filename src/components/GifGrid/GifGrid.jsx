import PropTypes from "prop-types";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h1>Cargando...</h1>}
      <div className="card-grid">
        {gifs.map((gif) => (
          <GifGridItem key={gif.id} title={gif.title} url={gif.url} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
