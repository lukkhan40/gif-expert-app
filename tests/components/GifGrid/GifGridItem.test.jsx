import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../../src/components/GifGrid/GifGridItem";

describe("Pruebas para <GifGridItem/>", () => {
  const title = "Bob 1";
  const url = "https://localhost:3000/";

  test("Evaluar el snapshot", () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Mostrar imagen con el URL y el alt indicado", () => {
    render(<GifGridItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
