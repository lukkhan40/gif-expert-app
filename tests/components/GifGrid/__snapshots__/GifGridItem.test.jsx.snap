// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGridItem/> Evaluar el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Bob 1"
      src="https://localhost:3000/"
    />
    <p>
      Bob 1
    </p>
  </div>
</div>
`;
